### PR TITLE
[1/2] Deployment Preparation: Yearn Linear Pool Factory

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Deployment preparation: `YearnLinearPoolFactory`
Build-info (linear-pools repo): [f2c98a31096b2c4b17106eec26334c4819191527](https://github.com/orbcollective/linear-pools/commit/f2c98a31096b2c4b17106eec26334c4819191527)

# Deployment preparation tasks checklist:

- [x] Deployments README is updated with the new task name and link
- [x] Task README has a description and links to the artifacts and expected output folders
- [x] build-info is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from build-info
- [x] New task has a working fork test

# Deprecating old tasks checklist:
 (Not appliable)

- [ ] Old task was moved to deprecated/ directory (already done)
- [ ] Deployments README is updated: old task is moved to the 'deprecated' section, and links are updated with the deprecated/ prefix
- [ ] Old task imports in index.ts, input.ts and fork tests are updated
- [ ] Old task README is updated with a warning sign, a link to the replacement task and a short description

# Code checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either master, or there's a description of how to merge

# Issue Resolution
N/A

### CODE REVIEW

**Comment by rabmarut** [comment]
Confirmed that I am also able to generate build-info `cacfa4f82d6bed0d5984ee9fc44ef4e6` out of commit `f2c98a31096b2c4b17106eec26334c4819191527` and that the file is identical to your `YearnLinearPoolFactory.json`.

File: pkg/deployments/tasks/20230213-yearn-linear-pool/test/contracts/ReadOnlyReentrancyAttackerYearnLP.sol [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        _vault = vault;
+    }
+
+    /**
+     * @dev Starts attack on target pool. Since LinearPool attacks are permissioned and regular joins and exits are
+     * disabled, the attacker contract must be the owner (or be granted permissions to perform the calls).
+     *
+     * It is possible to enter the Vault with a RecoveryMode exit - but `_handleRemainingEth` is only called if the
+     * Vault is requested to unwrap WETH, which requires a target pool that contains WETH.
+     * For the AAVE specialization the token pair should then be WETH and waWETH, but waETH doesn't seem to comply with
```
Comment (jubeira) [inline pkg/deployments/tasks/20230213-yearn-linear-pool/test/contracts/ReadOnlyReentrancyAttackerYearnLP.sol:53]: Same as in #2262 / https://github.com/balancer-labs/balancer-v2-monorepo/pull/2262#discussion_r1107548592:
if the contract is exactly the same, feel free to just delete this source code and use the one in the AAVE fork test for this task, adding a comment where you deploy the contract clarifying that the source is the same.

If this pool accepts WETH, then the attacker test for ERC4626 linear pool would be more suitable.

Finally, if the contract has any difference that I might be missing, just adjust these docs.

File: pkg/deployments/tasks/20230213-yearn-linear-pool/test/task.fork.ts [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+      const event = expectEvent.inReceipt(await tx.wait(), 'PoolCreated');
+
+      pool = await task.instanceAt('YearnLinearPool', event.args.pool);
+      expect(await factory.isPoolFromFactory(pool.address)).to.be.true;
+
+      poolId = await pool.getPoolId();
+      const [registeredAddress] = await vault.getPool(poolId);
+      expect(registeredAddress).to.equal(pool.address);
+
+      const { assetManager } = await vault.getPoolTokenInfo(poolId, USDC); // We could query for either frxEth or mooToken
```
Comment (jubeira) [inline pkg/deployments/tasks/20230213-yearn-linear-pool/test/task.fork.ts:152]:
```suggestion
      const { assetManager } = await vault.getPoolTokenInfo(poolId, USDC);
```
stale comment?

**Review by jubeira — COMMENTED** [review]
Build info looks correct, and fork test runs locally at the specified block number.

I've left a few minor comments; other than that, please remember to add the new task to `pkg/deployments/README.md`.

The rest is looking good!

File: pkg/deployments/README.md [hunk 3]
```diff
@@ -65,48 +65,49 @@ Returns an object with all contracts from a deployment and their addresses.
...
+| Gauge Registrant V3, supporting the Adaptor entrypoint | [`20230109-gauge-adder-v3`](./tasks/20230109-gauge-adder-v3)                                         |
+| Weighted Pool V3                                       | [`20230206-weighted-pool-v3`](./tasks/20230206-weighted-pool-v3)                                     |
+| Composable Stable Pools V3                             | [`20230206-composable-stable-pool-v3`](./tasks/20230206-composable-stable-pool-v3)                   |
+| Linear Pools for ERC4626 Tokens V3                     | [`20230206-erc4626-linear-pool-v3`](./tasks/20230206-erc4626-linear-pool-v3)                         |
+| Linear Pools for Aave aTokens (with rebalancing) V4    | [`20230206-aave-rebalanced-linear-pool-v4`](./tasks/20230206-aave-rebalanced-linear-pool-v4)         |
+| Linear Pools for Euler Tokens                          | [`20230208-euler-linear-pool`](./tasks/20230208-euler-linear-pool)                                   |
+| Linear Pools for Yearn Tokens                          | [`20230213-yearn-linear-pool`](./tasks/20230213-yearn-linear-pool)                                   |
```
Comment (jubeira) [inline pkg/deployments/README.md:110]: Same as in #2262; let's please just modify this line so that it doesn't cause extra collisions with other PRs.

**Review by jubeira — APPROVED** [review]
Looks good! Same comment as with #2262; otherwise it's good to go :)

File: pkg/deployments/README.md (hunk 3 above)
Comment (EndymionJkb) [reply pkg/deployments/README.md:110]: Tell me this isn't line endings again

**Comment by EndymionJkb** [comment]
These already include the build-info, so they're not [1/2]; can just be called "Deployment preparation: ..."

File: pkg/deployments/README.md (hunk 3 above)
Comment (jubeira) [reply pkg/deployments/README.md]: No, if they were it would show differently

Comment (jubeira) [reply pkg/deployments/README.md]: Also, DOS line endings were never commited anywhere :) it was just a 'local phenomenon'

Comment (joaobrunoah) [reply pkg/deployments/README.md]: @EndymionJkb @jubeira My IDE formats tables inside markdown files automatically. I'll disable that (My PC has a lot of surprises, ahahahhaha)

And about line endings, I changed my git configuration, it won't happen again, I'm being super careful about it :D

**Pull request merged**